File: VyacheslavBuchin/term2/block1/BMPFiltersApplication/src/main/java/bmp/BMPHeader.java
package bmp;

public record BMPHeader(
        char[] signature,
        int fileSize,
        int reserved,
        int dataOffset,

        int size,
        int width,
        int height,
        short planes,
        short bitsPerPixel,
        int compression,
        int imageSize,
        int xPixelsPerM,
        int yPixelsPerM,
        int colorsUsed,
        int colorsImportant
) {

    public BMPHeader {
        if (signature.length != 2 || signature[0] != 'B' || signature[1] != 'M')
            throw new IllegalArgumentException("Unsupported file format");
        if (bitsPerPixel != 24 && bitsPerPixel != 32)
            throw new IllegalArgumentException(String.format("Only 32 or 24 bits per pixel are supported. This file has %d", bitsPerPixel));
    }

}

File: Khelmianov-Artem/Term2/Block1/Filters/src/main/kotlin/bmp/BMPHeader.kt
package bmp


data class BMPHeader(
    val type: Short,
    val fileSize: Int,
    val reserved: Int,
    val offset: Int,

    val headerSize: Int,            /* Header size in bytes      */
    val width: Int,                 /* Width and height of image */
    val height: Int,
    val planes: Short,              /* Number of colour planes   */
    val bitsPerPixel: Short,        /* Bits per pixel            */
    val compression: Int,           /* Compression type          */
    val imageSize: Int,             /* Image size in bytes       */
    val xResolution: Int,           /* Pixels per meter          */
    val yResolution: Int,
    val numColors: Int,             /* Number of colours         */
    val numImportantColors: Int     /* Important colours         */
) {
    val isValid = type.toInt() == 19778 &&
            reserved == 0 &&
            planes.toInt() == 1 &&
            compression == 0 &&
            bitsPerPixel.toInt() in arrayOf(24, 32)
}
